### PR TITLE
Implement Generation Password and Compromised Checking Functionality for PassManager Android App

### PR DESCRIPTION
**Pull request by vita133**
This pull request adds the main functionality for password generation and the ability to check compromised passwords using the Have I Been Pwned API v3.

Key Changes:

1. Password Generation: Enhances the PasswordGenerator class to generate strong and secure passwords. Implements a randomized character selection algorithm based on user-defined criteria such as length and inclusion of character sets (numbers, lowercase letters, uppercase letters, special characters).

2. Checking Compromised Passwords: I created the PasswordChecker class that returns a Boolean answer to "Is the password compromised?". It uses the Have I Been Pwned API v3 to send requests and retrieve a list of similar password compromised hashes. The PasswordChecker is integrated into the PasswordGenerator and checks every generated password.

3. User Interface Updates: Modifies the user interface of the password generation screen (GenerationActivity) to display the strength of generated passwords. Fixes the view of elements in the login screen (LoginActivity).

4. Error Handling and Tests: Implements robust error handling and data validation mechanisms in the password generation process. Adds tests for the main scenarios of using the new features.

The application will assist users in creating stronger passwords and help them make informed decisions about their password security.

### CODE REVIEW

File: app/src/main/java/com/example/passmanager/login/model/generation/PasswordChecker.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                e.printStackTrace()
+            }
+
+            return@withContext false
+        }
+    }
+
+
+    private fun hashPassword(password: String): String {
+        val digest = MessageDigest.getInstance("SHA-1")
```
Comment (hrystynaa) [inline app/src/main/java/com/example/passmanager/login/model/generation/PasswordChecker.kt:57]: The code uses the deprecated SHA-1 hashing algorithm, which is no longer recommended for secure systems. It is better to use a more secure and modern hashing algorithm such as SHA-256.

```kotlin
val digest = MessageDigest.getInstance("SHA-256")
```

**Review by hrystynaa — COMMENTED** [review]
Here's a summary of the review:

1. It is better to update the code to use the SHA-256 algorithm for password hashing, which is a more secure choice compared to the previously used SHA-1. This change improves the overall security of the application.
2. You need to improve naming using camelCase conventional.

In conclusion, the code you provided is well-structured, readable, and follows good programming practices. Great work overall!

**Review by hrystynaa — COMMENTED** [review]
Here's a summary of the review:

1. It is better to update the code to use the SHA-256 algorithm for password hashing, which is a more secure choice compared to the previously used SHA-1. This change improves the overall security of the application.
2. You need to improve naming using camelCase conventional.

In conclusion, the code you provided is well-structured, readable, and follows good programming practices. Great work overall!

File: app/src/main/java/com/example/passmanager/login/model/generation/PasswordChecker.kt (hunk 1 above)
Comment (vita133) [reply app/src/main/java/com/example/passmanager/login/model/generation/PasswordChecker.kt:57]: You are right. I will do this!

File: app/src/main/java/com/example/passmanager/login/screens/GenerationActivity.kt [hunk 2]
```diff
@@ -1,21 +1,105 @@
...
     private lateinit var back: TextView
+    private lateinit var seekbarLength: SeekBar
+    private lateinit var edittextLength: EditText
+    private lateinit var checkboxNumbers: CheckBox
+    private lateinit var checkboxLowerCaseLetters: CheckBox
+    private lateinit var checkboxUpperCaseLetters: CheckBox
+    private lateinit var checkboxSymbols: CheckBox
+    private lateinit var buttonGenerate: Button
+    private lateinit var edittextPassword: EditText
+    private lateinit var textviewCompromised: TextView
```
[the remark this answers is not in the document]
Comment (vita133) [reply app/src/main/java/com/example/passmanager/login/screens/GenerationActivity.kt:27]: Thanks

File: app/src/main/java/com/example/passmanager/login/model/generation/PasswordChecker.kt (hunk 1 above)
Comment (vita133) [reply app/src/main/java/com/example/passmanager/login/model/generation/PasswordChecker.kt:57]: Sorry, I just checked that this API only provides a result list of hashes in SHA-1. That is why I must use this encrypted algorithm

**Pull request merged**